events/tutorials/tutorial_import_user_events_gcs.md

```diff
@@ -0,0 +1,185 @@
+# **Import User Events From the Cloud Storage Tutorial**
+
+## Let's get started
+
+The Retail API offers you an easy way to import your user event data from a Cloud Storage. All you need is to provide a name of
+the JSON file in the GCS bucket.
+
+This type of import is useful when you need to import a large amount of items to your catalog in a single step.
+
+You can find more information about different import types, their restrictions, and use cases in the [Retail API documentation](https://cloud.google.com/retail/docs/import-user-events)
+
+**Time to complete**: 
+<walkthrough-tutorial-duration duration="3.0"></walkthrough-tutorial-duration>
+
+## Before you begin
+
+To run Python code samples from this tutorial, you need to set up your virtual environment.
+
+To do that, run the following commands in a Terminal:
+```bash
+pip install virtualenv
+```
+```bash
+virtualenv <your-env>
+```
+```bash
+source <your-env>/bin/activate
+```
+Next, install Google packages:
+```bash
+pip install google
+```
+```bash
+pip install google-cloud-retail
+```
+
+**Tip**: Click the copy button on the side of the code box to paste the command in the Cloud Shell Terminal to
+run it.
+
+## Set the PROJECT_NUMBER environment variable
+
+As you are going to run the code samples in your own Cloud Project, you should specify the **project_id** as an environment variable. It will be used in every request to the Retail API.
+
+You can find the ```project_number``` in the **Home/Dashboard/Project Info card**.
+
+Set the environment variable with a following command:
+```bash
+export PROJECT_NUMBER=<YOUR_PROJECT_NUMBER>
+```
+
+## Upload user events data to the Cloud Storage bucket
+
+We have prepared a JSON file with a bunch of valid user events in the "events" directory: 
+
+**events/import_user_events_tutorial.json**
+
+You can use this file in the tutorial, or, if you want to use your own data, you should update the names of a bucket and a JSON file in the code samples.
+
+You should remember that you can only import to Retail catalog events which are **NOT older than 90 days** otherwise the import will fail.
```

Comment:
You can import events that are **NOT older than 90 days** into the Retail catalog. Otherwise, the import will fail.